src/tests/multihost/sssd/testlib/common/utils.py

```diff
@@ -1480,78 +1490,70 @@ def ad_conn(self):
         return ad_conn_inst
 
     def create_ad_unix_user_group(self, username, groupname,
-                                  mail=None, password='Secret123'):
+                                  mail=None, password='Secret123', uid=None):
         """ Create a AD User with Unix Attributes
 
         :param str username: AD User Name
         :param str groupname: AD Group Name
         :param str password: User password (default: Secret123)
+        :param int uid: User uid
         :Return bool: if user/group added correctly return True else False
         :Exceptions: False
         """
 
-        uid = random.randint(9999, 999999)
+        if uid is None:
+            uid = random.randint(9999, 999999)
         user_dn = "CN=%s,%s" % (username, self.ad_users_dn_entry)
         group_dn = "CN=%s,%s" % (groupname, self.ad_users_dn_entry)
-        cmd = self.ad_host.run_command(['dsadd.exe', 'user', user_dn, '-samid',
-                                        username, '-pwd', password])
-        cmd = self.ad_host.run_command(['dsadd.exe', 'group', group_dn])
+        self.ad_host.run_command(['dsadd.exe', 'user', user_dn, '-samid',
```

Comment:
Let's not use dsadd.exe instead user "powershell.exe Add-ADUser args", all the documentation for the ds commands is old, I can't seem to find any documentation post 2012r2 for these commands. I don't know if it's going to be deprecated, but I don't think we should take the chance. 

Can you update all the ds commands to use PowerShell, instead? https://docs.microsoft.com/en-us/powershell/module/activedirectory/set-aduser?view=windowsserver2022-ps

Reply:
Done. I really hope that the AD powershell did not change in a major way between windows versions, otherwise we might be in a world of pain as I did not go through every command and check it against documentation for all versions.